sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Helpers/SyntaxHelper.cs

```diff
@@ -181,9 +181,14 @@ public static bool IsAnyKind(this SyntaxToken syntaxToken, ICollection<SyntaxKin
         private static bool ContainsMethodInvocation(IEnumerable<SyntaxNode> syntaxNodes, SemanticModel semanticModel,
             Func<InvocationExpressionSyntax, bool> syntaxPredicate, Func<IMethodSymbol, bool> symbolPredicate)
         {
+            // See issue: https://github.com/SonarSource/sonar-csharp/issues/416

```

Comment:
// Where clause excludes nodes that are not defined on the same SyntaxTree as the
// SemanticModel (because of partial definition).

// Another approach would be to get the SemanticModel linked to the node we explore.
// More details: https://github.com/dotnet/roslyn/issues/18730